plugins/woocommerce/client/blocks/assets/js/blocks/product-filters/inner-blocks/taxonomy-filter/index.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { productFilterAttribute } from '@woocommerce/icons';
+import { getSetting } from '@woocommerce/settings';
+import { registerBlockType } from '@wordpress/blocks';
+import { __, sprintf } from '@wordpress/i18n';
+import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import Edit from './edit';
+import type { TaxonomyItem } from './types';
+
+const taxonomies = getSetting< TaxonomyItem[] >(
+	'filterableProductTaxonomies',
+	[]
+);
+
+registerBlockType( metadata, {
+	edit: Edit,
+	icon: productFilterAttribute,
+	save: () => {
+		return (
+			<div
+				{ ...useBlockProps.save() }
+				{ ...useInnerBlocksProps.save() }
+			/>
+		);
+	},
+	variations: taxonomies.map( ( item, index ) => {
+		return {
+			name: `product-filter-taxonomy-${ item.name }`,
+			title: item.label,
```

Comment:
I'd add `filter` suffix ASAP as inserter becomes confusing straight away 😅 

```suggestion
			title: `${ item.label } Filter`,
```

<img width="207" alt="image" src="https://github.com/user-attachments/assets/ffa20bdb-0a2b-4feb-86b7-8e2a62eede2c" />

<img width="222" alt="image" src="https://github.com/user-attachments/assets/d48c9b02-011a-4015-9562-5d5522cfe386" />

Reply:
Haha, yeah, I will do it. Attribute filter variations also need the same.